src/pages/Chat/Chat/index.tsx

```diff
@@ -11,153 +15,110 @@ import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 3,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(3),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 5,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(5),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 6,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(6),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 7,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(7),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 8,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(8),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 4,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(4),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 9,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(9),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 10,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(10),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 11,
-    text: 'Olá! Gostaria de adotar o seu gato!',
-    createdAt: new Date(11),
-    user: {
-      _id: 2,
-      name: 'React Native',
-    },
-  },
-  {
-    _id: 12,
-    text: 'Que maravilha, Marília! Você mora aqui em Brasília?',
-    createdAt: new Date(12),
-    user: {
-      _id: 1,
-      name: 'React Native',
-    },
-  },
-];
+import { styledComponents, styles } from './styles';
 
 export default (): JSX.Element => {
   const chatTitle = useRoute<RouteProp<RouteTypes.RouteParams, 'Chat'>>().params?.title;
-  const animalUID = useRoute<RouteProp<RouteTypes.RouteParams, 'Chat'>>().params?.animalUID;
   const targetUserUID = useRoute<RouteProp<RouteTypes.RouteParams, 'Chat'>>().params?.targetUserUID;
 
+  const routeChatUID = useRoute<RouteProp<RouteTypes.RouteParams, 'Chat'>>().params?.chatUID;
+  const currentUserRef = userAPI.currentUserDocument();
+
   useFocusEffect(
     useCallback(() => {
       setStatusBarBackgroundColor(Theme.elements.statusBarPrimaryDark, true);
     }, []),
   );
 
+  const navigation = useNavigation();
   const [messages, setMessages] = useState<IMessage[]>([]);
+  const [isLoadingEarlier, setIsLoadingEarlier] = useState<boolean>(false);
+  const [chatRef, setChatRef] = useState<DocumentRefData>();
+
+  const sorter = (a: IMessage, b: IMessage): number => (b.createdAt as Date).getTime() - (a.createdAt as Date).getTime();
+
+  const loadMessages = useCallback((lastMessage?: IMessage): void => {
+    if (!chatRef) return;
+
+    const lastTimestamp = lastMessage ? lastMessage.createdAt as Date : undefined;
```

Comment:
A complain on semantics, not logic: why is a variable of type Date called lastTimestamp and not lastDate or something along these lines?